yt_dlp/utils.py

```diff
@@ -666,6 +666,7 @@ def replace_insane(char):
     s = re.sub(r'[0-9]+(?::[0-9]+)+', lambda m: m.group(0).replace(':', '_'), s)  # Handle timestamps
     result = ''.join(map(replace_insane, s))
     if is_id is NO_DEFAULT:
+        result = result.encode(get_filesystem_encoding(), errors='ignore').decode(get_filesystem_encoding())
```

Comment:
shouldn't we do this in sanitize_filepath instead?

Reply:
I'm testing this using output template and it gets sanitized both ways, `sanitize_filename` is called through `prepare_filename` -> `evaluate_outtmpl`. 

Checking some of the usages `sanitize_path` might be better both because it doesn't handle ids and it handles the whole path.